### PR TITLE
[Java 5 warnings]

### PR DESCRIPTION
**Pull request by timtay-microsoft**
replace for loops with foreach loops, remove unnecessary boxing, replace a while loop with a foreach loop

### CODE REVIEW

**Review by vinagesh — APPROVED** [review]

File: device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/TransportClient.java [hunk 1]
```diff
@@ -146,9 +146,9 @@ public void open() throws IllegalStateException, IOException
     public void closeNow() throws IOException
     {
         // Codes_SRS_TRANSPORTCLIENT_12_015: [If the registered device list is not empty the function shall call closeFileUpload on all devices.]
-        for (int i = 0; i < this.deviceClientList.size(); i++)
+        for (DeviceClient deviceClient : this.deviceClientList)
```
Comment (abhipsaMisra) [inline device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/TransportClient.java:149]: +1

Comment (timtay-microsoft) [reply device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/TransportClient.java:149]: ?

**Review by abhipsaMisra — APPROVED** [review]

File: device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/TransportClient.java (hunk 1 above)
Comment (abhipsaMisra) [reply device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/TransportClient.java:149]: "I like it"

**Pull request merged**